### PR TITLE
#353, #252, #118 - view updates

### PR DESCRIPTION
**Pull request by kentmw**
added has and unset in view (for viewState) and added object mapping for prepareFields and _prepare option

### CODE REVIEW

File: modules/View.js [hunk 1]
```diff
@@ -145,6 +161,16 @@
     },
 
     /**
+     * Augments the context with custom content.
+     * @param context {Object} the context you can modify
+     * @return {Object} [Optional] If you return an object, it will be merged with the context
+     * @method _prepare
+     */
+    _prepare: function(context) {
```
Comment (mandragorn) [inline modules/View.js]: you could still use a _.noop here.

File: modules/View.js [hunk 2]
```diff
@@ -748,7 +782,11 @@
           }
         }
       }
-      return prepareFieldsContext;
+      var context = this._prepare(prepareFieldsContext);
+      if (!_.isUndefined(context)) {
```
Comment (mandragorn) [inline modules/View.js]: if _prepare() returns undefined then prepareFieldsContext is ignored.  I don't think that is what you want?

Please add tests for this so that we know it is working correctly for the corner cases and that it continues to work correctly :).

Comment (kentmw) [reply modules/View.js]: do the tests run for you? it breaks when i run them

Comment (kentmw) [reply modules/View.js]: also, good catch.

File: modules/View.js (hunk 1 above)
Comment (kentmw) [reply modules/View.js]: I know, but I thought that adding this empty function would be easier for people looking to see how _prepare works (i.e. there's a context argument

File: test/karma/viewPrepareSpec.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      });
+      var prepareView = new PrepareView();
+      var context = prepareView.prepare();
+      expect(context.foo).toBe(123);
+      expect(context.view).toBeDefined();
+      expect(context.view.bar).toBe(444);
+    });
+
+    it('can override existing context', function() {
+      var PrepareView = TorsoView.extend({
```
Comment (mandragorn) [inline test/karma/viewPrepareSpec.js]: should this set bar in view state to make sure it is being overwritten?  A little bit ambiguous since you are overwriting something that doesn't have anything set on it.  Not a huge deal.

File: test/karma/viewPrepareSpec.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      });
+      var prepareView = new PrepareView();
+      var context = prepareView.prepare();
+      expect(context.foo).toBe(555);
+      expect(context.view).toBeDefined();
+      expect(context.view.bar).toBe(444);
+    });
+
+    it('can return an overriding context', function() {
+      var PrepareView = TorsoView.extend({
```
Comment (mandragorn) [inline test/karma/viewPrepareSpec.js]: should this set bar in initialize?

File: test/karma/viewPrepareSpec.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        initialize: function() {
+          this.set('foo', 444);
+        },
+
+        _prepare: function(context) {
+          context.foo = 123;
```
Comment (mandragorn) [inline test/karma/viewPrepareSpec.js]: should this be context.view.foo?  To match with view.foo set in initialize?  Or do you want to add an expect that the context.view.foo is 444?

File: test/karma/viewPrepareSpec.js (hunk 3 above)
Comment (kentmw) [reply test/karma/viewPrepareSpec.js]: I'm overridding the whole view object.

File: test/karma/viewPrepareSpec.js (hunk 5 above)
Comment (kentmw) [reply test/karma/viewPrepareSpec.js]: yea, that wasn't great

File: test/karma/viewPrepareSpec.js (hunk 4 above)
Comment (kentmw) [reply test/karma/viewPrepareSpec.js]: i need to show this better
